### ISSUE TITLE
Add a `createAPIProject`

### ISSUE THREAD

**Issue by stephlocke** [lockedata/pRojects#22]
@hrbrmstr started a project a while ago close to my heart [hrbrmstr/swagger](https://github.com/hrbrmstr/swagger). 

I'd like to revive that project to generate functions from API definitions produced in swagger - this would enable us to point at a swagger definition and produce not just a package skeleton with best practices but functions too.

- [ ] Make sure swagger can generate functions from a definition
- [ ] Add dependency on swagger from here
- [ ] Write function that accepts a name and a swagger definition and generates a package project with R functions
- [ ] Extend the add-in to allow this type of project to be created

**Comment by maelle** [comment]
see also https://github.com/rstudio/swagger/issues/1#issuecomment-395627756 by @hrbrmstr too

**Comment by maelle** [comment]
see also https://github.com/mkearney/useapi by @mkearney

**Comment by maelle** [comment]
Any API examples for this @stephlocke?

**Comment by maelle** [comment]
this sure reminds me of our efforts in the hubspot package! Except Hubspot's API is not swagger compliant.

**Comment by stephlocke** [comment]
Every API is a special snowflake.
Stuff like adding the mocking infrastructure for tests, adding in ratelimitr etc might be a good start even if the dream would be to go from a swagger doc

**Comment by maelle** [comment]
Yeah, I'll revisit this once hubspot is in good shape.

Agree reg the dream. :sparkles:

**Comment by jonthegeek** [comment]
Did anything come of this? I'm trying to follow the breadcrumbs in various repo discussions 🙃

**Comment by maelle** [comment]
No, but for hubspot https://github.com/DoubleYouGTT/hubspot/ I vaguely remember writing code to automate adding new endpoints... although I see no trace of this now :thinking: Maybe the code only extracted endpoints to have a list of them, like a checklist. 

I think you had found https://blog.r-hub.io/2020/02/10/code-generation/ which is the only tip I'd have :sweat_smile:
